### PR TITLE
Cleanup cursor_demo.

### PR DESCRIPTION
**Pull request by anntzer**
- Less pyplot, more OO.
- Showcase both Cursor implementations.
- Misc. cleanups.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/misc/cursor_demo_sgskip.py [hunk 1]
```diff
@@ -3,19 +3,20 @@
...
+the figure with every mouse move.
 
 Faster cursoring is possible using native GUI drawing, as in
-wxcursor_demo.py.
+:doc:`/gallery/user_interfaces/wxcursor_demo_sgskip`.
 
 The mpldatacursor and mplcursors third-party packages can be used to achieve a
 similar effect.  See
 
     https://github.com/joferkington/mpldatacursor
```
Comment (timhoffm) [inline examples/misc/cursor_demo_sgskip.py]:
```suggestion
- https://github.com/joferkington/mpldatacursor
```

File: examples/misc/cursor_demo_sgskip.py [hunk 2]
```diff
@@ -3,19 +3,20 @@
...
     https://github.com/anntzer/mplcursors
```
Comment (timhoffm) [inline examples/misc/cursor_demo_sgskip.py]:
```suggestion
- https://github.com/anntzer/mplcursors
```

File: examples/misc/cursor_demo_sgskip.py [hunk 3]
```diff
@@ -3,19 +3,20 @@
...
+This example shows how to use Matplotlib to provide a data cursor.  It uses
+Matplotlib to draw the cursor and may be a slow since this requires redrawing
+the figure with every mouse move.
 
 Faster cursoring is possible using native GUI drawing, as in
-wxcursor_demo.py.
+:doc:`/gallery/user_interfaces/wxcursor_demo_sgskip`.
 
 The mpldatacursor and mplcursors third-party packages can be used to achieve a
 similar effect.  See
```
Comment (timhoffm) [inline examples/misc/cursor_demo_sgskip.py]: As far as I understand, the "similar" part is not quite true. This example focusses on drawing a cross-hair (free or snapped to data). The mentioned cursor packages add an annotation. I think the meaning of "cursor" is quite different here and there and thus should be clarified.

File: examples/misc/cursor_demo_sgskip.py [hunk 4]
```diff
@@ -3,19 +3,20 @@
...
 ===========
 
-This example shows how to use matplotlib to provide a data cursor.  It
-uses matplotlib to draw the cursor and may be a slow since this
-requires redrawing the figure with every mouse move.
+This example shows how to use Matplotlib to provide a data cursor.  It uses
+Matplotlib to draw the cursor and may be a slow since this requires redrawing
+the figure with every mouse move.
 
 Faster cursoring is possible using native GUI drawing, as in
```
Comment (timhoffm) [inline examples/misc/cursor_demo_sgskip.py]: There's also `https://matplotlib.org/gallery/widgets/cursor.html` which seems like a better builtin version of this example. It should at least be linked.

Given this, do we still need this example here? If so, it's probably only for educational puposes and that should be mentioned.

File: examples/misc/cursor_demo_sgskip.py (hunk 1 above)
Comment (anntzer) [reply examples/misc/cursor_demo_sgskip.py]: switched to rst anonymous link syntax

File: examples/misc/cursor_demo_sgskip.py (hunk 2 above)
Comment (anntzer) [reply examples/misc/cursor_demo_sgskip.py]: same as above.

File: examples/misc/cursor_demo_sgskip.py (hunk 3 above)
Comment (anntzer) [reply examples/misc/cursor_demo_sgskip.py]: Actually the wx example linked also does something different; the whole text could be redone.  Feel free to push if you have something better.

**Pull request merged**